.NET/Microsoft.Recognizers.Text.Number/Extractors/BaseNumberRangeExtractor.cs

```diff
@@ -143,9 +145,19 @@ private void GetMatchedStartAndLength(Match match, string type, string source, o
             else
             {
                 var numberStr = string.IsNullOrEmpty(numberStr1) ? numberStr2 : numberStr1;
-
+                
                 var extractNumList = ExtractNumberAndOrdinalFromStr(numberStr);
 
+                // If cases falls into "more/less than NUM" or "NUM or more/less"
+                if (extractNumList != null && (type == NumberRangeConstants.MORE || type == NumberRangeConstants.LESS))
```

Comment:
These two ifs should be combined into a method that determining if the special behaviour needs to be applied. If so, then it should tell ExtractNumberAndOrdinalFromStr which way to work.

Reply:
Fixed.